WCFDataService/Client/System/Data/Services/Client/QueryResult.cs

```diff
@@ -36,7 +36,7 @@ namespace System.Data.Services.Client
     /// <summary>
     /// Wrapper HttpWebRequest &amp; HttWebResponse
     /// </summary>
-    internal class QueryResult : BaseAsyncResult
+    internal class QueryResult : BaseAsyncResult, IDisposable
```

Comment:
Agreed. I do not like the addition of IDisposable to QueryResult. It is after all the the IODataResponseMessage we want to dispose, not the QueryResult it self. I will implement another solution for allowing the syncronous GetQuerySetCount ($count ) to dispose the message.